src/perception/long_range_cam/long_range_cam.cpp

```diff
@@ -0,0 +1,86 @@
+#include "long_range_cam.hpp"
+#include <opencv2/imgproc.hpp>
+#include <opencv2/videoio.hpp>
+#include <ros/init.h>
+#include <ros/this_node.h>
+#include <ros/time.h>
+#include <sensor_msgs/image_encodings.h>
+#include <thread>
+
+namespace mrover {
+
+    using namespace std::chrono_literals;
+
+    /**
+    * @brief Load config, open the camera, and start our threads
+    */
+    void LongRangeCamNodelet::onInit() {
+        try {
+            mGrabThread = std::jthread(&LongRangeCamNodelet::grabUpdate, this);
+        } catch (std::exception const& e) {
+            NODELET_FATAL("Exception while starting: %s", e.what());
```

Comment:
This won't ever get called. Starting a thread always works, and you can't catch exceptions that other threads throw. They have their own stack.